helper.go

```diff
@@ -181,6 +181,25 @@ func JsonMSet(client Client, ctx context.Context, kvs map[string]string, path st
 	return doMultiSet(client, ctx, cmds.s)
 }
 
+// DecodeSliceOfJSON is a helper that struct-scans each RedisMessage into dest, which must be a slice.
+func DecodeSliceOfJSON[T any](result RedisResult, dest *[]T) error {
+	values, err := result.ToArray()
+	if err != nil {
+		return err
+	}
+
+	ts := make([]T, len(values))
+	for i, v := range values {
+		var t T
+		if err = v.DecodeJSON(&t); err != nil && !IsRedisNil(err) {
+			return err
+		}
+		ts[i] = t
```

Comment:
```suggestion
func DecodeSliceOfJSON[T any](result RedisResult, dest *[]*T) error {
	values, err := result.ToArray()
	if err != nil {
		return err
	}

	ts := make([]*T, len(values))
	for i, v := range values {
		var t T
		if err = v.DecodeJSON(&t); err != nil && !IsRedisNil(err) {
			return err
		}
		ts[i] = &t
```

Considering the case of Nil, I think accepting a `*[]*T` is still a better approach.